### PR TITLE
DOI uploads fetch PDFs

### PR DESCRIPTION
**Pull request by lightninglu10**
Fetch PDFs on DOI upload

### CODE REVIEW

File: src/citation/constants.py [hunk 1]
```diff
@@ -156,6 +156,7 @@
         "issued",
         "note",
         "journal-name",
+        "pdf_url",
```
Comment (le-sun) [inline src/citation/constants.py:159]: Lets change this to `"URL"` because `"pdf_url"` is not CSL compliant.

File: src/citation/schema.py [hunk 2]
```diff
@@ -156,10 +156,11 @@ def generate_schema_for_citation(citation_type):
     "abstract": "abstract",
     "landing_page_url": "url",
     "primary_location": {
+        "pdf_url": "pdf_url",
```
Comment (le-sun) [inline src/citation/schema.py:159]: Related to above comment

File: src/utils/openalex.py [hunk 3]
```diff
@@ -1,9 +1,42 @@
...
+from paper.utils import get_pdf_from_url
 from researchhub.settings import OPENALEX_KEY
+from utils.aws import lambda_compress_and_linearize_pdf, upload_to_s3
+from utils.http import check_url_contains_pdf
 from utils.parsers import rebuild_sentence_from_inverted_index
 from utils.retryable_requests import retryable_requests_session
+from utils.sentry import log_error
+
+
+def download_pdf(url):
```
Comment (le-sun) [inline src/utils/openalex.py:14]: Let's move this function somewhere more appropriate

**Review by le-sun — CHANGES_REQUESTED** [review]

File: src/citation/constants.py (hunk 1 above)
Comment (lightninglu10) [reply src/citation/constants.py:159]: we already have another field "URL" -- what should we do about that other field?

Comment (lightninglu10) [reply src/citation/constants.py:159]: The CSL fields are slightly limiting, I'm thinking the best way we can handle it is if we can transform our own "fields" JSON into CSL fields with a function, that way we can discard fields when we need a citation but can keep other metadata.

Comment (le-sun) [reply src/citation/constants.py:159]: CSL has the additional `"custom"` field, where we can enter our own key-value pairs

Comment (lightninglu10) [reply src/citation/constants.py:159]: ah, we should add it in there then. Is "custom" a JSON field?

Comment (le-sun) [reply src/citation/constants.py:159]: Yep.
In `schema.py`
```
            "custom": {
                "title": "Custom key-value pairs.",
                "type": "object",
                "description": "Used to store additional information that does not have a designated CSL JSON field. The custom field is preferred over the note field for storing custom data, particularly for storing key-value pairs, as the note field is used for user annotations in annotated bibliography styles.",
                "examples": [
                    {"short_id": "xyz", "other-ids": ["alternative-id"]},
                    {"metadata-double-checked": True},
                ],
            },
```

Comment (lightninglu10) [reply src/citation/constants.py]: Okay, made the change here. Will review again and deploy tomorrow

**Pull request merged**